src/java.base/share/classes/sun/net/www/protocol/https/HttpsClient.java

```diff
@@ -434,6 +434,15 @@ protected Socket createSocket() throws IOException {
         }
     }
 
+    @Override
+    public void closeServer() {
+        try {
+            // SSLSocket.close may block up to timeout. Make sure it's short.
+            serverSocket.setSoTimeout(1);
+        } catch (Exception e) {}
```

Comment:
Is 1ms enough for a normal SSL close to complete? Does this mean that the normal close_notify exchange has to complete in that time?

Reply:
full close takes one RTT, which may be much more than 1 ms. However, we are not aiming for a full close here; [TLS spec](https://datatracker.ietf.org/doc/html/rfc5246#section-7.2.1) says we don't have to wait for the peer's close_notify after sending ours, and we don't wait for it.
The wait is to make sure the OS sends our close_notify message, and does not send a RST packet during that time if we happen to receive data from the peer. 1ms is definitely sufficient for HTTP 1, where the server should only send data when requested.